src/basic/mount-util.c

```diff
@@ -281,6 +281,15 @@ int path_is_mount_point(const char *t, const char *root, int flags) {
         if (path_equal(t, "/"))
                 return 1;
 
+        /* basename() does not work if the path has a trailing slash.
+         * Make sure it's not there. */
+        if (endswith(t, "/")) {
+                char *t2;
+
+                t2 = strdupa(t);
+                t = strip_trailing_chars(t2, '/');
+        }
```

Comment:
hmm, so with this change mounting something on /etc/fstab and then checking if its a mount point using /etc/fstab/ would return true...

On linux at least, access() and stat() do an implicitly directory check if you suffix a path with "/". i figure we should do the same here really, everything else is very likely to cause problems later on...

Reply:
I reworked our functions to preserve the trailing slash and let the kernel fail with -ENOTDIR.